Utility/WikiObjectPropertiesListGenerator.FCMacro

```diff
@@ -0,0 +1,401 @@
+__Web__ = ""
+__Wiki__ = "https://wiki.freecad.org/Macro_Wiki_Object_Properties_List_Generator"
+__Icon__ = "./WikiObjectPropertiesListGenerator.svg"
+__Xpm__ = ""
+__Help__ = "Select a scripted objects on the tree and execute the macro."
+__Status__ = "Beta"
+__Requires__ = ""
+__Communication__ = "https://forum.freecad.org/viewtopic.php?f=21&t=61998"
+__Files__ = "./WikiObjectPropertiesListGenerator.svg"
+
+# NOTE: Changelog
+# ---------
+# - 0.16 - 2024-11-15
+#   - Refactor code: add custom dataclass, update functions' signatures, make some fixes.
+#   - Include property's value on the list.
+#   - Create custom dialog to add checkbox to let user disable/enable property status.
+# - 0.15 - 2021-09
+#   - Initial release.
+
+import re
+
+import FreeCAD
+import FreeCADGui as Gui
+from PySide import QtCore, QtGui, QtWidgets
+
+from dataclasses import dataclass
+
+
+@dataclass
+class PropertyInfo:
+    property_type: str
+    data_type: str
+    default_value: str
+    tooltip: str
+    status: str
+
+
+# Based on https://github.com/FreeCAD/FreeCAD/blob/5d49bf78de785a536f941f1a6d06d432582a95d3/src/App/PropertyFile.h
+# and https://github.com/FreeCAD/FreeCAD/blob/5d49bf78de785a536f941f1a6d06d432582a95d3/src/App/PropertyContainerPyImp.cpp
+status_dictionary = {
+    0: "Touched",  # touched property
+    # 1:"Immutable",   # can't modify property
+    # 2:"ReadOnly",    # for property editor
+    # 3:"Hidden",      # for property editor
+    # 4:"Transient",   # for property container save
+    # 5:"MaterialEdit", # to turn ON PropertyMaterial edit
+    # 6:"NoMaterialListEdit", # to turn OFF PropertyMaterialList edit
+    # 7:"Output",      # same effect as Prop_Output
+    # 8:"LockDynamic", # prevent being removed from dynamic property
+    # 9:"NoModify",    # prevent causing Gui::Document::setModified()
+    # 10:"PartialTrigger", # allow change in partial doc
+    11: "NoRecompute",  # touch owner for recompute on property change
+    12: "Single",  # for save/load of floating point numbers
+    13: "Ordered",  # for PropertyLists whether the order of the elements is relevant for the container using it
+    14: "EvalOnRestore",  # In case of expression binding, evaluate the expression on restore and touch the object on value change.
+    # The following bits are corresponding to PropertyType set when the
+    # property added. These types are meant to be static, and cannot be
+    # changed in runtime. It is mirrored here to save the linear search
+    # required in PropertyContainer::getPropertyType()
+    21: "Dynamic",  # indicating the property is dynamically added
+    22: "NoPersist",  # corresponding to Prop_NoPersist
+    23: "NoRecompute",  # corresponding to Prop_NoRecompute
+    24: "ReadOnly",  # corresponding to Prop_ReadOnly
+    25: "Transient",  # corresponding to Prop_Transient
+    26: "Hidden",  # corresponding to Prop_Hidden
+    27: "Output",  # corresponding to Prop_Output
+}
+
+
+def clear_report_view():
```

Comment:
You should not clear the Report View without confirmation (I myself sometimes have some valuable information there). Add a warning in the appropriate dialog at least.